### PR TITLE
feat: add sub_bitmap

### PR DESCRIPTION
**Pull request by Mehrbod2002**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Add `sub_bitmap` function.

related to #11219

### CODE REVIEW

**Comment by ariesdevil** [comment]
Thanks for your contribution, please also add unit tests at https://github.com/datafuselabs/databend/tree/main/src/query/functions/tests/it/scalars

**Comment by Mehrbod2002** [comment]
@BohuTANG @ariesdevil 
You can review it now with the added tests .

File: src/query/functions/src/scalars/bitmap.rs [hunk 1]
```diff
@@ -143,6 +144,36 @@ pub fn register(registry: &mut FunctionRegistry) {
         ),
     );
 
+    registry.register_passthrough_nullable_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType, _, _>(
+        "sub_bitmap",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType>(
+            |b, item, length, builder, ctx| {
```
Comment (b41sh) [inline src/query/functions/src/scalars/bitmap.rs:151]: How about renaming `item` as `offset`?

Comment (Mehrbod2002) [reply src/query/functions/src/scalars/bitmap.rs]: Done

File: src/query/functions/src/scalars/bitmap.rs [hunk 2]
```diff
@@ -143,6 +144,36 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+    registry.register_passthrough_nullable_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType, _, _>(
+        "sub_bitmap",
+        |_, _, _| FunctionDomain::MayThrow,
+        vectorize_with_builder_3_arg::<BitmapType, UInt64Type, UInt64Type, BitmapType>(
+            |b, offset, length, builder, ctx| {
+                    match RoaringTreemap::deserialize_from(b) {
+                        Ok(rb) => {
+                            let subset_start = offset;
+                            let subset_length = length;
+                            if subset_start >= b.len() as u64 {
```
Comment (b41sh) [inline src/query/functions/src/scalars/bitmap.rs:156]: If the offset exceeds the length of the bitmap, we can return an empty bitmap without reporting an error.

File: src/query/functions/src/scalars/bitmap.rs [hunk 3]
```diff
@@ -143,6 +144,36 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+                                ctx.set_error(builder.len(), "invalid subset");
+                            } else {
+                                let adjusted_length = (subset_start + subset_length).min(b.len() as u64) - subset_start;
+                                let subset_bitmap = &rb.into_iter().collect::<Vec<_>>()[subset_start as usize ..(subset_start + adjusted_length) as usize];
+                                let mut rb = RoaringTreemap::new();
+                                for a in subset_bitmap.iter() {
+                                    rb.insert(*a);
```
Comment (b41sh) [inline src/query/functions/src/scalars/bitmap.rs:163]: better to use `RoaringTreemap::from_iter`

File: tests/sqllogictests/suites/query/02_function/02_0064_function_bitmap [hunk 4]
```diff
@@ -60,3 +60,8 @@ query TT
 SELECT bitmap_and_not(build_bitmap([2,3]), build_bitmap([2,3,5]))::String, bitmap_and_not(build_bitmap([1,3,5]), build_bitmap([1,5]))::String
 ----
 (empty) 3
+
+query F
+SELECT to_string(sub_bitmap(build_bitmap([1, 2, 3, 4, 5]), 1, 3))::String;
```
Comment (b41sh) [inline tests/sqllogictests/suites/query/02_function/02_0064_function_bitmap:65]: `::String` is an alias for `to_string`, just use one of them.

File: src/query/functions/src/scalars/bitmap.rs (hunk 2 above)
Comment (Mehrbod2002) [reply src/query/functions/src/scalars/bitmap.rs:156]: Done

File: src/query/functions/src/scalars/bitmap.rs [hunk 5]
```diff
@@ -143,6 +144,38 @@ pub fn register(registry: &mut FunctionRegistry) {
...
+                            let subset_length = length;
+                            if subset_start >= b.len() as u64 {
+                                let mut rb = RoaringTreemap::new();
+                                rb.serialize_into(&mut builder.data).unwrap();
+                                builder.commit_row();
+                            } else {
+                                let adjusted_length = (subset_start + subset_length).min(b.len() as u64) - subset_start;
+                                let subset_bitmap = &rb.into_iter().collect::<Vec<_>>()[subset_start as usize ..(subset_start + adjusted_length) as usize];
+                                let mut rb = RoaringTreemap::new();
+                                for a in RoaringTreemap::from_iter(subset_bitmap.iter()) {
```
Comment (b41sh) [inline src/query/functions/src/scalars/bitmap.rs:164]: using `RoaringTreemap::from_iter` can build a `RoaringTreemap`, so we can remove `RoaringTreemap::new` and for loop insert.

**Review by b41sh — APPROVED** [review]
LGTM

**Review by ariesdevil — APPROVED** [review]

**Comment by ariesdevil** [comment]
@mergifyio update

**Pull request merged**